sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -698,31 +722,35 @@ public void run() {
-                    StorageHelper.PreferencesStorage.putBoolean(PREF_KEY_ALWAYS_SEND, true);
-                }
-
-                Iterator<Map.Entry<UUID, ErrorLogReport>> unprocessedIterator = mUnprocessedErrorReports.entrySet().iterator();
-                while (unprocessedIterator.hasNext()) {
-                    if (shouldStopProcessingPendingErrors())
-                        return;
+                } else {
 
-                    Map.Entry<UUID, ErrorLogReport> unprocessedEntry = unprocessedIterator.next();
-                    ErrorLogReport errorLogReport = unprocessedEntry.getValue();
-                    ErrorAttachment attachment = mCrashesListener.getErrorAttachment(errorLogReport.report);
-                    if (attachment == null)
-                        MobileCenterLog.debug(LOG_TAG, "CrashesListener.getErrorAttachment returned null, no additional information will be attached to log: " + errorLogReport.log.getId().toString());
-                    else
-                        errorLogReport.log.setErrorAttachment(attachment);
-                    mChannel.enqueue(errorLogReport.log, ERROR_GROUP);
+                    if (userConfirmation == ALWAYS_SEND) {
+                        StorageHelper.PreferencesStorage.putBoolean(PREF_KEY_ALWAYS_SEND, true);
+                    }
 
-                    /* Clean up an error log file and map entry. */
-                    unprocessedIterator.remove();
-                    ErrorLogHelper.removeStoredErrorLogFile(unprocessedEntry.getKey());
+                    Iterator<Map.Entry<UUID, ErrorLogReport>> unprocessedIterator = mUnprocessedErrorReports.entrySet().iterator();
+                    while (unprocessedIterator.hasNext()) {
+                        if (shouldStopProcessingPendingErrors())
+                            break;
+
+                        Map.Entry<UUID, ErrorLogReport> unprocessedEntry = unprocessedIterator.next();
+                        ErrorLogReport errorLogReport = unprocessedEntry.getValue();
+                        ErrorAttachment attachment = mCrashesListener.getErrorAttachment(errorLogReport.report);
+                        if (attachment == null)
+                            MobileCenterLog.debug(LOG_TAG, "CrashesListener.getErrorAttachment returned null, no additional information will be attached to log: " + errorLogReport.log.getId().toString());
+                        else
+                            errorLogReport.log.setErrorAttachment(attachment);
+                        mChannel.enqueue(errorLogReport.log, ERROR_GROUP);
+
+                        /* Clean up an error log file and map entry. */
+                        unprocessedIterator.remove();
+                        ErrorLogHelper.removeStoredErrorLogFile(unprocessedEntry.getKey());
+                    }
                 }
+
+                /* Processed crash report for the last session. */
+                if (isInstanceEnabled())
+                    mHandler.getLooper().quit();
```

Comment:
No. It goes through initialization steps but it won't post any runnables to that handler because the service already deleted all pending files when it disabled.